components/team_sidebar/team_button/team_button.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Link} from 'react-router-dom';
+import {Team} from 'mattermost-redux/types/teams';
+
+type Props = {
+    team: Team;
+};
+
+type State = {
+
+};
+
+export default class TeamButton extends React.PureComponent<Props, State> {
```

Comment:
Nothing to act now but in general we can just use functions for components as small as this. React components are expensive in general and adding a new one for small ones like this can cause more harm than good.

Reply:
Agreed, we're not currently using this component anyway but I plan to change a few of these components to FCs at some point.